common/flatpak-run.c

```diff
@@ -1460,7 +1462,13 @@ flatpak_run_add_environment_args (FlatpakBwrap    *bwrap,
         }
     }
 
-  flatpak_context_append_bwrap_filesystem (context, bwrap, app_id, app_id_dir, previous_app_id_dirs, &exports);
+
+  exports = flatpak_context_get_exports_full (context,
+                                              app_id_dir, previous_app_id_dirs,
+                                              TRUE, TRUE,
+                                              xdg_dirs_conf, &home_access);
```

Comment:
This would look cleaner to me if xdg_dirs_conf was created by flatpak_context_get_exports_full and returned as an out-value. Its kinda weird how you need to pass it an empty GString which gets filled in.

Reply:
Thanks, I'll take a look at that. Any thoughts on #4214? That's the other branch that I need to land to have Steam's subsandboxing work.

Reply:
I was copying the interface of the pre-existing function `flatpak_context_export()`, which already took an empty GString and appended to it, but you're right that an "out" argument feels more GLib-ish.

I'm adding a new commit that changes both `flatpak_context_export()` and `flatpak_context_get_exports_full()` to have an "out" argument, `char **xdg_dirs_conf_out`.